src/uu/kill/src/kill.rs

```diff
@@ -177,23 +182,31 @@ fn list(arg: Option<String>) {
     };
 }
 
-fn kill(signalname: &str, pids: &[String]) -> i32 {
-    let mut status = 0;
+fn kill(signalname: &str, pids: &[String]) -> UResult<()> {
     let optional_signal_value = uucore::signals::signal_by_name_or_value(signalname);
     let signal_value = match optional_signal_value {
         Some(x) => x,
-        None => crash!(EXIT_ERR, "unknown signal name {}", signalname),
+        None => {
+            return Err(USimpleError::new(
+                1,
+                format!("unknown signal name {}", signalname),
+            ));
+        }
     };
     for pid in pids {
         match pid.parse::<usize>() {
             Ok(x) => {
                 if unsafe { libc::kill(x as pid_t, signal_value as c_int) } != 0 {
-                    show_error!("{}", Error::last_os_error());
-                    status = 1;
+                    return Err(USimpleError::new(1, format!("{}", Error::last_os_error())));
```

Comment:
`last_os_error` also returns an `io::Error` so again `UIoError` can be used, although it doesn't add much in this case. Up to you how you want to do it.